test_conformance/conversions/basic_test_conversions.cpp

```diff
@@ -326,8 +327,10 @@ int CalcRefValsPat<InType, OutType, InFP, OutFP>::check_result(void *test,
         for (uint32_t i = 0; i < count; i++)
             if (t[i] != c[i] &&
                 // Allow nan's to be binary different
-                !((t[i] & 0x7fff) > 0x7C00 && (c[i] & 0x7fff) > 0x7C00)
-                && !(a[i] != (cl_uchar)0 && t[i] == (c[i] & 0x8000)))
+                !((t[i] & 0x7fff) > 0x7C00 && (c[i] & 0x7fff) > 0x7C00) &&
+                !(a[i] != (cl_uchar)0 && t[i] == (c[i] & 0x8000)) &&
+                // retry per section 7.5.3.2, with ftz on
+                !(IsHalfSubnormal(c[i]) && (HTF(t[i]) == 0.0f)))
```

Comment:
The reasoning behind this change seems good but the fix seems slightly wrong.  In short, the `a` array is supposed to track whether the input may be flushed to zero or not, so this case _should_ be handled by the preceding line, which tests if the input may be flushed to zero (`a[i] != (cl_uchar)0`) and then if it actually is flushed to zero, with the proper sign (`t[i] == (c[i] & 0x8000)`).

It looks like there is a problem with the way the `a` array is setup though, for `half` inputs (or outputs).  Specifically, I think we need to define a specific `setAllowZ` for `half`, and call it here:

```c++
        if (gForceFTZ)
        {
            if (inType == kfloat || outType == kfloat)
                setAllowZ((uint8_t *)a, (uint32_t *)s, count);
        }
        if (gForceHalfFTZ)
        {
            if (inType == khalf || outType == khalf)
                setAllowZ((uint8_t *)a, (uint32_t *)s, count);   // <-- needs a half-specific version
        }
```

If this is done then I think this specific change can be reverted.  Can you give this a try and see if it fixes your issue?  Thanks!